client/state/checkout/emergent-paywall/reducer.js

```diff
@@ -0,0 +1,27 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import { EMERGENT_PAYWALL_RECEIVE } from 'state/action-types';
+
+/**
+ * Returns the updated emergent paywall iframe configuration state after an action has been dispatched.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object} Updated state
+ */
+export default function( state = {}, action ) {
+	switch ( action.type ) {
+		case EMERGENT_PAYWALL_RECEIVE:
+			return {
+				chargeId: action.charge_id,
+				payload: action.payload,
+				signature: action.signature,
+				url: action.paywall_url,
```

Comment:
it'd be great if we didn't propagate snake_case names into the app. it'd also be nice to avoid `_RECEIVE` nomenclature.

we can fix this by providing an API transformer in the data layer handler and keep the API mapping concern separate from the internal state concern

```js
const fromApi = apiData => ( {
	chargeId: apiData.charge_id,
	payload: apiData.payload,
	signature: apiData.signature,
	url: apiData.paywall_url,
} );

dispatchRequestEx( {
	fetch,
	onSuccess: ( action, { chargeId, payload, signature, url } ) => ( {
		// replace with call to `configureEmergentyPaywall()`
		type: EMERGENT_PAYWALL_IFRAME_CONFIG,
		chargeId,
		payload,
		signature,
		url
	} ),
	onError: noop, // what happens if it fails? customer waits forever?
	fromApi, // transforms response before sending data into `onSuccess`
} )
```

now our reducer can look quite semantic.